.github/workflows/pull_request.yml

```diff
@@ -53,6 +53,19 @@ jobs:
         run: |
           git --no-pager diff
 
+      - name: Build P4 constants
+        id: check
+        run: |
+          make p4-constants
+          echo "::set-output name=PORCELAIN::`git status --porcelain`"
+      # Verify P4 constants
+      - name: Check P4 constants are updated
+        if: ${{ steps.check.outputs.PORCELAIN != '' }}
+        uses: actions/github-script@v3
+        with:
+          script: |
+            core.setFailed('Please run make p4-constants and commit changes')
```

Comment:
If this is best practice for GHA then it's fine.

Here's a very simple way of checking for modifications with git: https://github.com/stratum/stratum/blob/main/.circleci/check-cpp-format.sh#L268-L269

Prints the list of files and set the bash status code to 1 if updates are needed.

Reply:
I don't think it's a best practice. Using now `git update-index --refresh` instead of `git status --porcelain`